binfinder.py

```diff
@@ -5,13 +5,99 @@
 logger = logging.getLogger(__name__)
 
+
+#----------------------------------------------------------------------
+def add_prepfold_to_commands(commands, pointing, pulsar, obsid, nbins, use_mask=True):
+        
+    comp_config = config.load_config_file()
+    #Figure out whether or not to input a mask
+    if use_mask == True:
+        check_mask = glob.glob("{0}{1}/incoh/*.mask".format(comp_config['base_product_dir'], obsid))
+        if check_mask:
+            mask = "-mask " + check_mask[0]
+        else:
+            mask = ""
+    else:
+        mask=""
+    
+    #find the beginning and end of the pulsar's beam coverage for this obs
+    start, end = pulsar_beam_coverage(obsid, pulsar) 
+    logger.info("start and end of pulsar beam coverage for on-disk files:{0}, {1}".format(start, end))
+    if start>=1.:
+        logger.error("pulsar is not in beam for any of the on-disk files. Ending...")
+        sys.exit(1)
+
+    #load presto module here because it uses python 2
+    commands.append('cd {0}'.format(pointing))
+    commands.append('echo "Folding on known pulsar {0}"'.format(pulsar))
+    commands.append('psrcat -e {0} > {0}.eph'.format(pulsar))
+    commands.append("sed -i '/UNITS           TCB/d' {0}.eph".format(pulsar))
+    commands.append("prepfold -o {0}_{2}_bins -noxwin -nosearch -runavg -noclip -timing {1}.eph"\
+                    " -nsub 256 -n {2} {3} -start {4} -end {5} 1*.fits"\
+                    .format(obsid, uplsar, nbins, mask, start, end))
+    commands.append('errorcode=$?')
+    commands.append('pulsar={}'.format(pulsar[1:]))
+    pulsar_bash_string = '${pulsar}'
+    #Some old ephems don't have the correct ra and dec formating and
+    #causes an error with -timing but not -psr
+    commands.append('if [ "$errorcode" != "0" ]; then')
+    commands.append('   echo "Folding using the -psr option"')
+    commands.append('   prepfold -o {0}_{2}_bins -noxwin -nosearch -runavg -noclip -psr {1}'\
+                    ' -nsub 256 -n {2} {3} -start {4} -end {5} 1*.fits'\
+                    .format(obsid, pulsar, nbins, mask, start, end))
+    commands.append('   pulsar={}'.format(pulsar))
+    commands.append('fi')
+    commands.append('rm {0}.eph'.format(pulsar))
+
+    return commands
+
+#----------------------------------------------------------------------
+def pulsar_beam_coverage(obsid, pulsar):
+    #returns the beginning and end time as a fraction that a pulsar is in the primary beam for the obsid files
+
+    #find the enter and exit times of pulsar normalized with the observing time
+    names_ra_dec = fpio.grab_source_alog(pulsar_list=[pulsar])
+    beam_source_data, _ = fpio.find_sources_in_obs([obsid], names_ra_dec)
+
+    enter_obs_norm = beam_source_data[obsid][0][1]
+    exit_obs_norm = beam_source_data[obsid][0][2]
+
+    #find the beginning and end time of the observation FILES you have on disk
+    files_beg, files_end = checks.find_beg_end(obsid)
+    files_duration = files_end - files_beg
```

Comment:
why?

Reply:
Math, if I have start at 1 and end at 2 then 2 - 1 + 1 = the two seconds of data I have

Reply:
so if your beginning and end have the same value then you still have one second of data and not zero? Which means that the beginning is taken as the beginning of the integer and the end is actually the end of the integer? are you sure that's how it works?

Reply:
So think of it like using the combined files which have 1 file for each second of data. What you're really saying is what is the first data file and the last data file. If in your example it's only one second long then the start and stop would be the same time. Using your equation it would be x-x = 0 which is incorrect because you are using one second of data

Reply:
Okay, that makes sense. Will change